src/Lib/Managers/ModeManager.vala

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2021 Alecaddd (http://alecaddd.com)
+ *
+ * This file is part of Akira.
+ *
+ * Akira is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * Akira is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with Akira. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Martin "mbfraga" Fraga <mbfraga@gmail.com>
+ */
+
+/**
+ * Manages InteractionMode's via registration. This class can be plugged into the event methods of a Canvas
+ * in order to allow these modes to affect items in the canvas as well as allowing them to absorb said
+ * events.
+ *
+ * Only one InteractionMode is guaranteed to exist at a time, and this manager will correctly alert the
+ * beginning and end of a registered mode.
+ *
+ * The exception to the above rule is pan_mode, which can be running on top of another mode in certain cases and
+ * masks events appropriately.
+ *
+ * See ItneractionMode.vala for more details on how to create modes.
```

Comment:
nit: little typo in the InteractionMode.

Reply:
done